src/data/value/literal.rs

```diff
@@ -41,6 +42,10 @@ impl PartialEq<Literal<'_>> for Value {
                 Err(_) => false,
             },
             (Value::Interval(l), Literal::Interval(r)) => l == r,
+            (Value::UUID(l), Literal::Text(r)) => match parse_uuid(r) {
+                Ok(r) => l == &r,
+                Err(_) => false,
+            },
```

Comment:
This is a comment on `parse_uuid`.
How about improving it this way?

```suggestion
            (Value::UUID(l), Literal::Text(r)) => Uuid::parse_str(v).map(Uuid::as_u128).map(|r| l == &r).unwrap_or(false)
```